Replace `done` callbacks in the font-tests with async/await instead

Snuffleupagus opened the pull request.

Snuffleupagus committed:
Replace `done` callbacks in the font-tests with async/await instead

Snuffleupagus commented:
/botio fonttest

Snuffleupagus commented:
/botio fonttest

timvandermeij commented on test/font/fontutils.js:
```diff
@@ -65,24 +65,26 @@ function encodeFontData(data) {
-  const xhr = new XMLHttpRequest();
-  xhr.open("POST", "/ttx");
+function ttx(data) {
+  return new Promise((resolve, reject) => {
```
Can't we just make this function `async` instead so we don't need to wrap it in a `Promise` manually?

Snuffleupagus replied:
Unfortunately I cannot see how that'd work, given that `xhr.onreadystatechange` is a *callback* function that we cannot simply `await` :-(

We could *maybe* re-factor this function to use `fetch` instead, assuming this isn't using e.g. a `file:`-URL, but that's probably better left for a follow-up.

Snuffleupagus replied:
It seems that using `fetch` instead worked just fine, so I've pushed a separate commit that fixes this :-)

timvandermeij reviewed (commented):
Looks good to me with the comment addressed.

Snuffleupagus committed:
Convert the font-test `ttx` helper function to use the Fetch API

By replacing `XMLHttpRequest` with a `fetch` call, the helper function can be modernized to use async/await instead.
Note that the headers doesn't seem necessary to set now, since:
 - The Fetch API provides a method for accessing the response as *text*, which renders the "Content-type" header unnecessary.
 - According to https://developer.mozilla.org/en-US/docs/Glossary/Forbidden_header_name, the "Content-length" header isn't necessary.

Snuffleupagus commented:
/botio fonttest

timvandermeij commented:
Looks really good, and using the fetch API is also much cleaner. Thank you for doing this!
